tests/test_app.py

```diff
@@ -70,3 +86,35 @@ def test_call(self):
         response = func(request)
         assert response.status == '200 OK'
         assert response.text == '9'
+
+    if PY2:
+        def test_generate_signature(self):
+            def sample_function(x, one="hey", **kwargs):
+                pass
+            func = FireflyFunction(sample_function)
+            assert len(func.sig) == 3
+            assert func.sig[0]['name'] == 'x'
+            assert func.sig[0]['kind'] == 'POSITIONAL_OR_KEYWORD'
+            assert func.sig[1]['name'] == 'one'
+            assert func.sig[1]['kind'] == 'POSITIONAL_OR_KEYWORD'
+            assert func.sig[1]['default'] == 'hey'
+            assert func.sig[2]['name'] == 'kwargs'
+            assert func.sig[2]['kind'] == 'VAR_KEYWORD'
+    if PY3:
+        def test_generate_signature_py3(self):
+            # work-around to avoid syntax error in python 2
+            code = 'def f(x, y=1, *, one="hey", **kwargs): pass'
+            env = {}
+            exec(code, env, env)
+            f = env['f']
+
+            func = FireflyFunction(f)
+            assert len(func.sig) == 4
+            assert func.sig[0]['name'] == 'x'
+            assert func.sig[0]['kind'] == 'POSITIONAL_OR_KEYWORD'
+            assert func.sig[1]['default'] == 1
+            assert func.sig[2]['name'] == 'one'
+            assert func.sig[2]['kind'] == 'KEYWORD_ONLY'
+            assert func.sig[2]['default'] == 'hey'
+            assert func.sig[3]['name'] == 'kwargs'
+            assert func.sig[3]['kind'] == 'VAR_KEYWORD'
```

Comment:
Use [pytest.skipif][1] decorator instead of if condition like this. Give different names for those two functions. `_py2` and `_py3` suffix would probably be better.

[1]: https://docs.pytest.org/en/latest/skipping.html#id1